spikes/lk_imu_kit/main.cpp

```diff
@@ -34,20 +34,22 @@ IMUKit imukit(imu::lsm6dsox);
 
 auto main() -> int
 {
+	rtos::ThisThread::sleep_for(140ms);
 	logger::init();
 
 	HelloWorld hello;
 	hello.start();
 
 	imu::lsm6dsox.init();
 
+	imukit.stop();
 	imukit.init();
 	imukit.start();
 
+	const auto &[pitch, roll, yaw] = imukit.getEulerAngles();
+
 	while (true) {
-		auto [pitch, roll, yaw] = imukit.getAngles();
 		log_info("Pitch : %7.2f, Roll : %7.2f Yaw : %7.2f", pitch, roll, yaw);
```

Comment:
This is a funny thing. Sonarcloud says to use a const reference to avoid unnecessary copying of the values.

Doing so, we are able to call the function only once and still have the values updated in the while loop.

Not sure it's a good thing though: one might want to store the data and use the values later than ~20ms after the function call. In this case, the values will not hold the old ones, but new ones.

So Sonarcloud is not right there. It depends on the context. Maybe getEulerAngles should not return a reference but the struct to avoid the issue in any case.

Reply:
> one might want to store the data and use the values later than ~20ms after the function call. In this case, the values will not hold the old ones, but new ones
> Maybe getEulerAngles should not return a reference but the struct to avoid the issue in any case.

I don't see current application of a such behavior but this being said, and according to Motion Kit function's `run()` current implementation (getEulerAngles in a while loop), it might be better to return the struct.

Reply:
This has been implemented with no issues and is much less error prone